include/boost/config/detail/suffix.hpp

```diff
@@ -528,12 +528,6 @@ namespace boost {
 #  define BOOST_APPEND_EXPLICIT_TEMPLATE_NON_TYPE(t, v)
 #  define BOOST_APPEND_EXPLICIT_TEMPLATE_NON_TYPE_SPEC(t, v)
 
-// When BOOST_NO_STD_TYPEINFO is defined, we can just import
-// the global definition into std namespace:
-#if defined(BOOST_NO_STD_TYPEINFO) && defined(__cplusplus)
-#include <typeinfo>
-namespace std{ using ::type_info; }
-#endif
```

Comment:
I have to manually delete this block of code to get Boost.Config to compile with the IAR compiler, so I hope this PR gets merged soon!

Reply:
As I mentioned on the issue tracker, I'm not sure removing these lines is the best solution after all.

Reply:
This was an MSVC-specific workaround as I recall, if I restrict the code to MSVC does that help?

BTW is there really no typeinfo?  What else is missing?

Reply:
Really and truly. Granted, this is _because_ I have disabled exceptions and RTTI on this compiler because our target is embedded. So for example I have `BOOST_NO_EXCEPTIONS`, `BOOST_NO_RTTI` and `BOOST_NO_TYPEID` defined in the preprocessor, and that solves most anticipated problems.